File: backend/pkg/testdata/gfmt/interactive-guess/solutions/ok.go
package main

import (
	"fmt"
)

func main() {
	var n int
	if _, err := fmt.Scan(&n); err != nil {
		return
	}
	low, high := 1, n
	for low <= high {
		mid := (low + high) / 2
		fmt.Printf("? %d\n", mid)
		var response string
		if _, err := fmt.Scan(&response); err != nil {
			break
		}
		if response == "=" {
			fmt.Printf("! %d\n", mid)
			break
		} else if response == ">" {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
}

File: backend/pkg/testdata/gfmt/interactive-guess/solutions/ok.py
import sys

def main():
    line = sys.stdin.readline()
    if not line:
        return
    n = int(line.strip())
    
    low, high = 1, n
    while low <= high:
        mid = (low + high) // 2
        print(f"? {mid}", flush=True)
        
        response = sys.stdin.readline()
        if not response:
            break
        response = response.strip()
        
        if response == "=":
            print(f"! {mid}", flush=True)
            break
        elif response == ">":
            low = mid + 1
        else:
            high = mid - 1

if __name__ == '__main__':
    main()
